### ISSUE TITLE
Text in brackets sometime prevents recognition of the following tag

### ISSUE THREAD

**Issue by aisbergde** [mpcjanssen/simpletask-android#1204]
try this content

```
2022-09-25 +test (Text in brackets sometimes prevents recognition of the following tag) +test2 +test3
2022-09-25 +test (Text in brackets with following dot). +test3
2022-09-25 +test (Text in brackets without following dot) +test3

```

the +test2 is not recognized as tag. I tried to find out the logic, when this happens, but I didn't. I think it doesn't depend on the dot.

you can also remove the +test3 after +test2 and it will not be recognized.

![Screenshot_2022-09-25-08-20-10-592_nl.mpcjanssen.simpletask.jpg](https://user-images.githubusercontent.com/9020187/192131033-95956171-9e84-4d54-8799-f053186be06b.jpg)

**Comment by aisbergde** [comment]
as a work around I can add one more space in front of the tag. But I also use the same todo files on Windows (I use a GitLab repository to sync). But the Windows app sleek can't keep the original order of lists and tags and resorts them.

I need to find a way to use Simpletask on Windows.

**Comment by mpcjanssen** [comment]
For me the tags are recognized and higlighted properly:

![image](https://user-images.githubusercontent.com/146895/192134235-13c6f423-77b4-4664-871e-1fb0f2574d02.png)
![image](https://user-images.githubusercontent.com/146895/192134251-274dcde7-f748-43b0-927b-40c920977451.png)

Which version are you using? (Find at top of Settings -> Other -> View debug log)

**Comment by aisbergde** [comment]
Simpletask cloudless v11.0.1

btw, I can't copy and paste content from the log. Please implement a way to easy copy the current version. See (can't find the other issue here on Android, because GitHub is always opening in this terrible mobile app where it is not possible to do anything else in parallel)

**Comment by aisbergde** [comment]
as workaround I can add an additional blank before the '+test2', but when I open the todo. txt on Windows in sleek, then unfortunately this program rearranges all content.

**Comment by sphenotux** [comment]
Just guessing; check linebreak, carriage return, vbcrlf, EOL whatever on Windows vs. Android/Linux?
If linebreak is NOT recognised, maybe due to wrong settings of the file, two 'separate' tasks may be interpreted as ONE by ST. Where did you create the text file, which machine?

**Comment by aisbergde** [comment]
I am on Android. I found this issue today when I copied some content from a forum into simpletask. I created 2 tasks, added a tag using Simpletask. And then I wanted to filter by this tag and only one task got the tag. I found out, that it is somehow related to the parentheses, I could reproduce with one test task, but I don't get the issue on some other test tasks. I can't find the logic. It is something with the parser. But the issue can be reproduced. I can create another file, just copied only the relevant content.

So what I posted is already the extract of some testing.

It could be special characters, which are not visible on Android.

So I will check after synchronize with the PC, if there is something not shown in Android.

**Comment by aisbergde** [comment]
I think there is an invisible special character:
when I just copy I can reproduce the issue.
when I then remove the visible space before the +test2 and add it again, then the +test2 is a task.

So it is solved.

But this could happen again, when a user will copy and past content from wherever. Maybe it is possible to implement a function to clean up content and to remove non visible characters?

**Comment by sphenotux** [comment]
Hm... Remove? Or replace? If the latter, replace by... What? Single whitespace?
On the other hand, it is IMHO not up to ST cleaning up the mess someone else from far away created 😉, in your case the (unknown) author of the source you copied from.

**Comment by smichel17** [comment]
> I can't copy and paste content from the log

You can share from the log, along with something like [share to clipboard](https://f-droid.org/en/packages/com.tengu.sharetoclipboard/)

**Comment by smichel17** [comment]
As for the issue: the best outcome imo would not be to automatically sanitize, but just *warn* the user that there is some character which may not be render on Android / might cause parsing issues. The warning can show only on the add task screen, so it isn't too annoying for people who might want to use these characters.

**Comment by aisbergde** [comment]
Yes, a warning would be good. Especially when you are on Android, you think that what you see is what it is. So you don't notice such errors. On the PC, there are a few tools that can clean up the content of the clipboard so that only the pure text is copied. On Android, you have to take what's available.

By the way, I haven't found out on the PC what kind of special character might have been hidden there.

**Comment by aisbergde** [comment]
I can't upload, but I used a hex editor to check the content

![Screenshot_2022-09-27-22-41-44-486_nl mpcjanssen simpletask](https://user-images.githubusercontent.com/9020187/192631181-4d70316a-c666-4f62-8fcc-bd865225b1ce.jpg)
![Screenshot_2022-09-27-22-41-23-090_com myprog hexedit](https://user-images.githubusercontent.com/9020187/192631211-02494b1a-d6b6-4baa-bb31-ffe035ee3df9.jpg)

c2 a0

**Comment by aisbergde** [comment]
https://www.scivision.dev/hex-code-c2a0-non-breaking-space-markdown/

Hex C2A0 is UTF8 for non-breaking space

**Comment by mpcjanssen** [comment]
@aisbergde thanks for the feedback. Could you try with one of the versions from https://mpcjanssen.nl/artifacts/11.0.1-5-bd94175/ ?

**Comment by aisbergde** [comment]
I will do. But it looks like I need to uninstall my current f-droid version first. And there is no backup and restore for settings, so I will do some Screenshots.

Maybe it would be a good idea to enable not only save and restore of filters, but also for settings?

Or I will try to install on another smartphone, where the settings are not so important.

**Comment by aisbergde** [comment]
it looks better now, regarding test2
but there is a strange first empty task
Maybe, this is not new but was already in the current version

![Screenshot_2022-09-28-00-03-33-375_nl.mpcjanssen.simpletask.jpg](https://user-images.githubusercontent.com/9020187/192645992-29260da6-8425-446a-8487-027eb902aa36.jpg)

**Comment by mpcjanssen** [comment]
You can install multiple version of Simpletask without issue. So just try one of the debug builds and you won't have to uninstall the FDroid version.

**Comment by aisbergde** [comment]
I can't
Otherwise I would not write that I can't.

![Screenshot_2022-09-28-08-03-58-255_com.google.android.packageinstaller.jpg](https://user-images.githubusercontent.com/9020187/192700249-f6117143-6a22-4438-9025-884390f87d10.jpg)

![Screenshot_2022-09-28-08-04-06-639_com.google.android.packageinstaller.jpg](https://user-images.githubusercontent.com/9020187/192700268-a6f63a4b-6daf-4dad-8166-0ede1a01e931.jpg)

it is in conflict with an existing installation.
I also tried the other applications, but they can't open local files or they are looking for encrypted files.

**Comment by aisbergde** [comment]
but it looks like that I can install the debug version in parallel. Then I misunderstood the instruction "any version", I should try any debug version. 🤔

**Comment by smichel17** [comment]
Yeah, you can install the different backend "flavors" of SimpleTask alongside each other, and for each "flavor", you can install the release and debug version alongside each other. You cannot install the release version of the same flavor over the one from F-Droid.

On September 28, 2022 6:12:18 AM UTC, "Germo Görtz" ***@***.***> wrote:
>but it looks like that I can install the debug version in parallel. Then I misunderstood the instruction "any version", I should try any debug version. 🤔
>
>
>
>-- >
>Reply to this email directly or view it on GitHub:
>
>https://github.com/mpcjanssen/simpletask-android/issues/1204#issuecomment-1260434788
>
>You are receiving this because you commented.
>
>
>
>Message ID: ***@***.***>
-- 
To respect your time, I try to write short, functional emails.
